rhythm/calendardelay.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package rhythm
+
+import "time"
+
+// CalendarDelaySchedule represents a recurring duty cycle that includes days, months or years too, e.g. "Every 3 months".
+// It does not support jobs more frequent than once a second.
+type CalendarDelaySchedule struct {
+	years  int
+	months int
+	days   int
+
+	Delay time.Duration
+}
+
+func EveryCalendar(years, months, days int, duration time.Duration) CalendarDelaySchedule {
```

Comment:
Pass values as struct with named fields. Multiple parameters with the same type end up as bugs.

Reply:
This is used internally only, but added an extra struct. It is still possible to instantiate the struct without naming the fields, though. So, it is up to the caller in the end.